### ISSUE TITLE
Monitoring UI does not allow wildcards in index names

### ISSUE THREAD

**Issue by Jon-AtAWS** [opendistro-for-elasticsearch/community#34]
Creating an alert, in the Define Monitor portion of the page, you specify the index. When I start typing, I get a drop down with possible choices that are specific indexes. I want to run the alert over an index pattern, like logs-*. I need to have that choice in the menu. 

If, I type logs-* into the text box, and escape from the drop down, then the UI errors, asking me to set an index name.

<img width="420" alt="image" src="https://user-images.githubusercontent.com/48454554/55364830-01e39e00-5497-11e9-81d1-e85c77d6596c.png">

**Comment by lucaswin-amzn** [comment]
Hi Jon, please take a look at the [answer here](https://discuss.opendistrocommunity.dev/t/cant-use-index-pattern-when-defining-monitor/320).

When completing the index name, hit enter and the UI will no longer show an error.

**Comment by Jon-AtAWS** [comment]
Can you add what the user has typed so far, with a trailing * to the dropdown in position 0? Then if I click that I won't see the error. 

Or remove the error if I typed a wildcard that matches one or more of my indexes.

**Comment by dbbaughe** [comment]
Hi Jon,

Did this help text not show up for you?
<img width="587" alt="Screen Shot 2019-04-01 at 4 44 55 PM" src="https://user-images.githubusercontent.com/46505179/55366433-9c46e000-549d-11e9-8c9c-a9e5c3dbba5e.png">

**Comment by Jon-AtAWS** [comment]
Better!

It would be great to avoid the error state entirely, since it's not really an error.

**Comment by dbbaughe** [comment]
Hi Jon,

In this case, typing and hitting esc is an error since you did not actually select (or create) an option and this is field requires at least one option selected. Perhaps it's a bit confusing since it's not just a text input box, but rather a [combo box](https://elastic.github.io/eui/#/forms/combo-box) that Elastic has created. This would be the equivalent of selecting an input and then moving on without typing anything.  We can definitely make the error text more helpful in this situation so the user can easily remedy the problem.

**Comment by Jon-AtAWS** [comment]
This is a minor point, and I'm happy with the solution proposed: better error text and messaging in the combo box to hit enter.

To nitpick, though, if I've typed something then I have "made a selection". The widget should allow either user-entered text or a selection from the attached menu. Minor point, but I disagree that typing something and hitting escape is an error on my part. The text box on Kibana's "Management" page for selecting an index pattern works this way - it's a text box and Kibana also gives me an auto-updated list of possible matches I can click on. It would be more consistent to have that experience.

**Comment by dbbaughe** [comment]
Yes, that is true for Kibana's "Management" page because it only allows you to select one index. This UX for Alerting allows you to select multiple indices which means we need you to explicitly select the indices. If this looks to still be an issue after better messaging then perhaps we can revisit with a different UI component.

Thanks!
